pkg/shell/hosts_dialog.jsx

```diff
@@ -247,7 +255,7 @@ class AddMachine extends React.Component {
         const title = this.state.old_machine ? _("Edit host") : _("Add new host");
         const submitText = this.state.old_machine ? _("Set") : _("Add");
 
-        const body = <Form isHorizontal>
+        const body = <Form isHorizontal onSubmit={prevent_default(callback)}>
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19018-20230627-115453-74cfaf23-fedora-38-devel-storage//Coverage/lcov/github-pr.diff.gcov.html)